crates/sargon/src/signing/collector/signatures_collector_preprocessor.rs

```diff
@@ -29,7 +29,53 @@ pub(crate) fn sort_group_factors(
         .collect::<IndexSet<_>>()
 }
 
+pub struct ExtractorOfInstancesRequiredToSignTransactions;
+impl ExtractorOfInstancesRequiredToSignTransactions {
```

Comment:
Let's move
This into a separate file, and add docs and a few unit test for this specifically ? You can use our manifests builders to build manifests requiring auth.

Use the sample values of the SignaturesCollector with securified accounts examples.

Michael created a function which allows you to create manifests requiring auth of personas and accounts